conwayste/src/uilayout.rs

```diff
@@ -0,0 +1,126 @@
+/*  Copyright 2019 the Conwayste Developers.
+ *
+ *  This file is part of conwayste.
+ *
+ *  conwayste is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  conwayste is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with conwayste.  If not, see
+ *  <http://www.gnu.org/licenses/>. */
+
+use std::collections::HashMap;
+
+use ggez::graphics::{Rect, Font};
+
+use ggez::{Context};
+
+use crate::constants::{
+    widget_ids::*,
+};
+use crate::config::Config;
+use crate::Screen;
+use crate::ui::{
+    Widget,
+    Button,
+    Checkbox,
+    Chatbox,
+    Layer,
+    Pane,
+    TextField,
+    UIAction,
+    helpe,
+};
+
+pub struct UILayout {
+    pub layers: HashMap<Screen, Vec<Layer>>,
+}
+
+impl UILayout {
```

Comment:
I thought the idea would be that everything in the `ui/` folder would be reusable components, without specifying any particular layout of the components. All this stuff setting up widgets and their positions would be better in `client.rs` or somewhere else outside of the `ui/` folder IMHO.

Reply:
This file isn't in the `src/ui/` folder. It's in `src/`.